x/ibctransfermiddleware/keeper/keeper.go

```diff
@@ -104,3 +109,111 @@ func (k Keeper) GetChannelFeeAddress(ctx sdk.Context, targetChannelID string) st
 	}
 	return channelFee.FeeAddress
 }
+
+type BridgeFee struct {
+	Fee      sdk.Coin
+	Sender   sdk.AccAddress
+	Receiver sdk.AccAddress
+}
+
+func (k Keeper) ChargeFee(ctx sdk.Context, msg *ibctypes.MsgTransfer) (*BridgeFee, error) {
```

Comment:
Why `ChargeFee` has something to do with timeouts?

Reply:
I will rename function
thanks

Reply:
Renamed.